POL-217: Add Syntax examples for Condition builder

spinder opened the pull request:
I'm not getting the final spacing around the StackItem 'right', too much space, or able to figure out how to get the drop down flag widget moved to the left of the accordian display.  
![Screenshot2020-04-17_17-57-12](https://user-images.githubusercontent.com/1906187/79617226-f597a680-80d4-11ea-9747-0722df84bcc4.png)
![Screenshot2020-04-17_17-56-35](https://user-images.githubusercontent.com/1906187/79617227-f597a680-80d4-11ea-9ea0-0c7c4b70a4a5.png)

josejulio commented:
I see that you used the [accordion](https://www.patternfly.org/v4/documentation/react/components/accordion) component, but the [Expandable](https://www.patternfly.org/v4/documentation/react/components/expandable)  (I think you can use the Uncontrolled one) matches design better.

There are conflicts now

spinder commented:
![Screenshot2020-04-17_22-45-22](https://user-images.githubusercontent.com/1906187/79626334-618e0500-80fd-11ea-9cd4-e9f9207d8d40.png)
![Screenshot2020-04-17_22-44-33](https://user-images.githubusercontent.com/1906187/79626335-62269b80-80fd-11ea-8af7-d39928422107.png)

0) Rebased to get rid of merge errors.
1)Used the new component and removed all the other code related to the previous component and state tracking.
2)The spacing inside the Expandable is bigger than I thought it would be. Suggestions? Should we put a small gray box around it when expanded? The issue is that fully expanded the conditions box is pushed off the screen.  Go still smaller on the text size? Decrease spacing in between lines?

josejulio commented:
I think we need to put a left padding ([see mock up](https://user-images.githubusercontent.com/1906187/79626334-618e0500-80fd-11ea-9cd4-e9f9207d8d40.png)) of the expanded content (but not the expandable link it self) to align it properly

> 1. 2)The spacing inside the Expandable is bigger than I thought it would be. Suggestions? Should we put a small gray box around it when expanded? The issue is that fully expanded the conditions box is pushed off the screen.  Go still smaller on the text size? Decrease spacing in between lines?

The spacing inside seems too much for me, in the design looks good because the space is too big  there :)

@katierik what do you suggest? should we pull the condition text above the text or?

pilhuhn commented:
How does that look scrolled down? I.e. when the examples are expanded and I want to finally write the rule? Should the rule input field always be visible and only the examples scrolling (if expanded)?

katierik commented:
To make sure I understand - 
1 - We cannot achieve the spacing around the expandable header
![image](https://user-images.githubusercontent.com/23172038/79758395-58628b00-82eb-11ea-92ad-708eb6c017be.png)

2 - We cannot indent the content in
![image](https://user-images.githubusercontent.com/23172038/79758440-6adcc480-82eb-11ea-8b3b-6ec681086658.png)

?

josejulio commented:
> 1 - We cannot achieve the spacing around the expandable header
> 2 - We cannot indent the content in

I think we can do that, @spinder can you apply these changes?

Also, for the monospaced fonts, we can use the font-family 'monospace'.
To do that we can create a monospace className with typestyle and use `fontFamily: 'monospace'` and apply that className where required, see https://github.com/RedHatInsights/policies-ui-frontend/blob/8ae293685763bb102bb257518201a66920bb82d3/src/components/Policy/Table/ExpandedContent/Actions.tsx#L22-L24 and https://github.com/RedHatInsights/policies-ui-frontend/blob/8ae293685763bb102bb257518201a66920bb82d3/src/components/Policy/Table/ExpandedContent/Actions.tsx#L93

katierik commented:
We can probably reduce the spacing more with the text styles
![image](https://user-images.githubusercontent.com/23172038/79773424-733efa80-82ff-11ea-9e19-37dc27b2d73c.png)

spinder commented:
![Screenshot2020-04-20_14-25-27](https://user-images.githubusercontent.com/1906187/79787157-8d370800-8314-11ea-963a-07f31b1b6f56.png)

I think I've made the suggested changes, but it still a bit too big(IMO) to see the Conditions after expanding.  @josejulio, @katierik ? Wdyt.

spinder commented:
![Screenshot2020-04-20_17-28-51](https://user-images.githubusercontent.com/1906187/79801944-0e9a9480-832d-11ea-954b-432b94c6bd60.png)
Indented the usage area.

josejulio commented on src/components/Policy/Table/ExpandedContent/Usage.tsx:
```diff
@@ -0,0 +1,51 @@
+    marginBottom: 5
+});
+const padLeftClassName = style({
+    marginLeft: 12
```
Could you bump this to 20?

josejulio commented:
With the typeahead on conditions, the options are hard to use / see when "condition examples" are unrolled.
When I try to scroll, it loses it focus and the suggestions box closes.

Should we move the condition text to the top? or what we can do in this case?

![Peek 2020-04-21 11-41](https://user-images.githubusercontent.com/3845764/79890373-02faac80-83c5-11ea-8bd0-400b050bfcc5.gif)
@katierik

spinder commented:
![Screenshot_2020-04-21_14-10-58](https://user-images.githubusercontent.com/1906187/79899166-6d6a1780-83da-11ea-8218-61aa3e0c4309.png)
![Screenshot_2020-04-21_14-10-28](https://user-images.githubusercontent.com/1906187/79899167-6d6a1780-83da-11ea-9bbc-cd7dde18a7f8.png)
![Screenshot2020-04-21_14-09-05](https://user-images.githubusercontent.com/1906187/79899168-6e02ae00-83da-11ea-8bd6-3e435d9a0dcb.png)
![Screenshot_2020-04-21_14-08-14](https://user-images.githubusercontent.com/1906187/79899170-6e02ae00-83da-11ea-8cda-67c2d7a99fdc.png)
@josejulio, does this work? I did some quick checking and it looks like the widgets continue to work correctly. Please give it the once over..
